src/main/java/org/broadinstitute/hellbender/tools/pon/coverage/CoveragePoNQCUtils.java

```diff
@@ -24,47 +23,41 @@
+     * @param qcPoN -- a QC PoN, which typically has fewer eigensamples.
      * @param ctx -- use null if not using Spark
      * @return never {@code null}
      */
-    public static List<String> retrieveSamplesWithArmLevelEvents(final File qcPonFilename, JavaSparkContext ctx) {
-        List<String> result;
-        try (final HDF5File ponReader = new HDF5File(qcPonFilename, HDF5File.OpenMode.READ_ONLY)) {
-
-            logger.info("Loading existing PoN...");
-            final PoN qcPoN = new HDF5PoN(ponReader);
-
-            // For each sample in the PoN, tangent normalize against the qc reduced PoN.
-            logger.info("Tangent Normalizing all normals in the PoN...");
-            final TangentNormalizationResult qcTangentNormalization = TangentNormalizer.tangentNormalizeNormalsInPoN(qcPoN, ctx);
-
-            // Get the median target CR estimate per contig across all samples.
-            final Map<String, Double> contigToMedian = getContigToMedianCRMap(qcTangentNormalization.getTangentNormalized());
-
-            // Segment the result
-            // For each tangent normalization result from the QC PoN
-            if (ctx == null) {
-                final List<ReadCountCollection> singleSampleTangentNormalizedReadCounts = createIndividualReadCountCollections(qcTangentNormalization.getTangentNormalized());
-                result = identifySamplesWithSuspiciousContigs(singleSampleTangentNormalizedReadCounts, contigToMedian);
-            } else {
-                final JavaRDD<ReadCountCollection> parallelSingleSampleTangentNormalizedReadCounts = createParallelIndividualReadCountCollections(qcTangentNormalization.getTangentNormalized(), ctx);
-                result = identifySamplesWithSuspiciousContigs(parallelSingleSampleTangentNormalizedReadCounts, ctx, contigToMedian);
-            }
+    public static <T extends CoveragePoNNormalizationResult> List<String> retrieveSamplesWithArmLevelEvents(final CoveragePanelOfNormals<T> qcPoN, final JavaSparkContext ctx) {
+        Utils.nonNull(qcPoN, "QC PoN cannot be null.");
+        final List<String> result;
+
+        // For each sample in the PoN, tangent normalize against the qc reduced PoN.
+        logger.info("Tangent normalizing all normals in the PoN...");
+        final CoveragePoNNormalizationResult qcNormalization = qcPoN.normalizeNormalsInPoN(ctx);
+
+        // Get the median target CR estimate per contig across all samples.
+        final Map<String, Double> contigToMedian = getContigToMedianCRMap(qcNormalization.getProfile());
+
+        // Segment the result
+        // For each tangent normalization result from the QC PoN
+        if (ctx == null) {
+            final List<ReadCountCollection> singleSampleTangentNormalizedReadCounts = createIndividualReadCountCollections(qcNormalization.getProfile());
+            result = identifySamplesWithSuspiciousContigs(singleSampleTangentNormalizedReadCounts, contigToMedian);
```

Comment:
Instead of declaring `result` without initializing above, why not just `return identifySamplesWith. . .` here and below?